tests/compatibility/BasicTestCase.nut

```diff
@@ -58,6 +58,30 @@ class BasicTestCase extends ImpTestCase {
         }.bindenv(this));
     }
 
+	function testSendDefault() {
+        return Promise(function(resolve, reject) {
+            local mm = MessageManager({
+                "firstMessageId":  msgId,
+                "nextIdGenerator": msgIdGenerator
+            });
+            mm.onReply(function(msg, response) {
+                try {
+                    assertDeepEqualWrap(DEFAULT_MESSAGE, response.data, ERR_REQ_RES_NOT_IDENTICAL);
+                    resolve();
+                } catch (ex) {
+                    reject(ex);
+                }
+            }.bindenv(this));
+            mm.onFail(function(msg, reason, retry) {
+                reject("onFail handler called. Reason: " + reason);
+            });
+            mm.onTimeout(function(msg, wait, fail) {
+                fail();
```

Comment:
If we reject everywhere else, why should we do fail with no helpful information here?
Let's do reject with some comments to help developers find out what's happening in case it fails this way.

Reply:
Currently, we have reject in onFail handler with "User called fail" reason when we do fail().
I agree, it would be better and more informative to use  reject("onTimeout handler called") instead fail(), but in this case we need replace fail() in all existing tests. Will we do it?